docs/adding_packages/sources_and_patches.md

```diff
@@ -6,28 +6,195 @@ These are a very important aspects and it helps us to establish the quality of t
+which are enforced by the [linters](../../linter/conandata_yaml_linter.py).
+
+Patches must be located in the recipe folder in a `patches/` sub-directory.
+
+There are a few recommendations about naming patches:
+
+* be descriptive but terse
+* number them so they can be re-used
+* note the specific version
+
+By clearly indicating what the patch does, when it's applied, and how it relates to existing patches- you can
+help make the [review process](../review_process.md) easier for readers and help speed up your pull requests.
+
+### Exporting Patches
+
+It's ideal to minimize the number of files in a package to exactly what's required. When recipes support multiple
+versions with differing patches it's strongly encouraged to only export the patches that are being used for that given recipe.
+
+Make sure the `export_sources` attribute is replaced by
+[`conan.tools.files.export_conandata_patches`](https://docs.conan.io/en/latest/reference/conanfile/tools/files/patches.html?highlight=export_conandata_patches)
+helper.
+
+```py
+def export_sources(self):
+    export_conandata_patches(self)
+```
+
+### Applying Patches
+
+Patches can be applied in a separate method, the pattern name is `_patch_sources`. When applying patch files,
+using [`conan.tools.files.apply_conandata_patches`](https://docs.conan.io/en/latest/reference/conanfile/tools/files/patches.html?highlight=apply_conandata_patches)
+is the best option.
+
+```py
+def build(self):
+    apply_conandata_patches(self)
+```
+
+For more complicated cases,
+[`conan.tools.files.rm`](https://docs.conan.io/en/latest/reference/conanfile/tools/files/basic.html#conan-tools-files-rm)
+or [`conan.tools.files.replace_in_file`](https://docs.conan.io/en/latest/reference/conanfile/tools/files/basic.html#conan-tools-files-replace-in-file)
+good choices.
+
+```py
+def _patch_sources(self):
+    # remove bundled libfmt
+    rmdir(self, os.path.join(self.source_folder, "lib", "fmt"))
+    replace_in_file(self, os.path.join(self.source_folder, "CMakeLists.txt"), "${CMAKE_SOURCE_DIR}", "${CMAKE_CURRENT_SOURCE_DIR}")
+```
+
+### Rules
+
+These are the rules that apply to regular version of Conan packages:
+
+**Build system patches.** In order to add libraries into ConanCenter sometimes
+it is NEEDED to apply patches so they can consume existing packages
+for requirements and binaries can be generated. These patches are totally
+needed for the purpose of ConanCenter and Conan keeps adding features trying
+to minimize these changes.
+
+**Source patches.** ConanCenter DOES NOT accept patches **backporting bugfixes or
+features** from upcoming releases, they break the principle of minimum surprise,
+they change the behavior of the library and it will no longer match the
+documentation or the changelog originally delivered by the authors.
+
+However, ConanCenter DOES accept **working software patches**, these patches
```

Comment:
maybe, `working software` needs some strict definition. as it might be too subjective what is considered working or non-working.

Reply:
I agree -- I think it's worth revisiting this to more accurately capture the types of patches we have been seeing since this policy came out :)